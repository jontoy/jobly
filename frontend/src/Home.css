.Home {
    height: 95vh;
    font-size: 1.5rem;
}
.Home h1{
    margin-top: 20%;
    font-size: 4rem;
}