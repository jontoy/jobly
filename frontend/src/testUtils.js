import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
