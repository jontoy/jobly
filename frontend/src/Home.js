import React, {useContext} from 'react';
import UserContext from './userContext';
import './Home.css';

const Home = () => {
    const {currentUser} = useContext(UserContext);
    return (
        <div className="Home jumbotron">
            <h1 className="display-4">Jobly</h1>
            <p className="lead">All the jobs in one, convenient place.</p>
            {currentUser ? 'Welcome Back!' : null}
        </div>
    )
}

export default Home;