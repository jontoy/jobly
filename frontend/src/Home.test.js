import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(
        <UserProvider>
            <Home />
        </UserProvider>
    );
});
  
it('matches snapshot when user is logged in', () => {
      const {asFragment} = render(
            <UserProvider>
                <Home />
            </UserProvider>
        );
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when user is logged out', () => {
    const {asFragment} = render(
          <UserProvider currentUser={null}>
              <Home />
          </UserProvider>
      );
  expect(asFragment()).toMatchSnapshot();
});