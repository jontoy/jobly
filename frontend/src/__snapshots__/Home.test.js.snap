// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="Home jumbotron"
  >
    <h1
      class="display-4"
    >
      Jobly
    </h1>
    <p
      class="lead"
    >
      All the jobs in one, convenient place.
    </p>
    Welcome Back!
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user is logged out 1`] = `
<DocumentFragment>
  <div
    class="Home jumbotron"
  >
    <h1
      class="display-4"
    >
      Jobly
    </h1>
    <p
      class="lead"
    >
      All the jobs in one, convenient place.
    </p>
  </div>
</DocumentFragment>
`;
